hw/ip/otbn/doc/dv_plan/index.md

```diff
@@ -24,83 +16,124 @@ applicable. Once done, remove this comment before making a PR. -->
+The idea is that this is much quicker for designers to use to sanity-test proposed changes, and can be run with Verilator, so it doesn't require an EDA tool licence.
+This pre-DV phase cannot drive sign-off, but it does use much of the same tooling.
+
+Once we are running full DV tests, we re-use this work, by using the same collection of randomised instruction streams and randomly picking from them for most of the sequences.
+The ELF binaries are all placed in a single directory and named sequentially (`0.elf`, `1.elf`, ..., `1234.elf` or similar).
+This directory and the maximum index are passed to the simulation with plusargs.
+
+<div class="bd-callout bd-callout-warning">
+
+**TODO:** There's a bit of infrastructure work to do here to teach `dvsim.py` to run the instruction generator. Also, maybe it would be better to pass a list of ELF files (which would let us filter the initial generation for things that hit coverage points in pre-DV). This design might change slightly.
+
+</div>
+
+The pre-DV testing doesn't address external stimuli like resets or TileLink-based register accesses.
+These are driven by specialised test sequences, described below.
+
 #### Test sequences
-All test sequences reside in `hw/ip/otbn/dv/uvm/env/seq_lib`.
-The `otbn_base_vseq` virtual sequence is extended from `cip_base_vseq` and serves as a starting point.
-All test sequences are extended from `otbn_base_vseq`.
-It provides commonly used handles, variables, functions and tasks that the test sequences can simple use / call.
-Some of the most commonly used tasks / functions are as follows:
-* task 1:
-* task 2:
+
+The test sequences can be found in `hw/ip/otbn/dv/uvm/env/seq_lib`.
+The basic test sequence (`otbn_base_vseq`) loads the instruction stream from a randomly chosen binary (see above), configures OTBN and then lets it run to completion.
+
+More specialized sequences include things like multiple runs, register accesses during operation (which should fail) and memory corruption.
+We also check things like the correct operation of the interrupt registers.
 
 #### Functional coverage
-To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
-The following covergroups have been developed to prove that the test intent has been adequately met:
-* cg1:
-* cg2:
+
+<div class="bd-callout bd-callout-warning">
+
+**TODO**: Functional coverage points are not yet defined.
+
+</div>
+
 
 ### Self-checking strategy
 #### Scoreboard
-The `otbn_scoreboard` is primarily used for end to end checking.
-It creates the following analysis ports to retrieve the data monitored by corresponding interface agents:
-* analysis port1:
-* analysis port2:
-<!-- explain inputs monitored, flow of data and outputs checked -->
+
+If any of the consistency checks in `otbn_core_model` that compare the RTL with the ISS fail, the module raises an error signal, bound as `err_if`.
```

Comment:
Is it required to have it connected to the scoreboard? Seems like an assertion check in the model ought to be sufficient...

Reply:
Yeah, good point. That's much simpler.